### PR TITLE
fix _asdict

### PR DESCRIPTION
**Pull request by Ed-XCF**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,5 +15,7 @@ pytz==2021.1
 redis==3.5.3
 sqla-wrapper==4.200628
 SQLAlchemy==1.3.19
+pytz==2021.1
```
Comment (JoshYuJump) [inline requirements.txt:18]: `pytz` has included in line 14

File: requirements.txt [hunk 2]
```diff
@@ -15,5 +15,7 @@ pytz==2021.1
...
+python-dateutil==2.8.2
```
Comment (JoshYuJump) [inline requirements.txt:19]: python-dateutil is included by `dateparser==1.0.0`

File: bali/utils/timezone.py [hunk 3]
```diff
@@ -75,5 +77,28 @@ def localtime(value: datetime = None, timezone: StrTzInfoType = None):
     return value.astimezone(timezone)
 
 
-def localdate(value: datetime = None, timezone: StrTzInfoType = None):
+def localdate(value: datetime = None, timezone: StrTzInfoType = None) -> date:
     return localtime(value, timezone).date()
+
+
+def start_of(
```
Comment (JoshYuJump) [inline bali/utils/timezone.py:84]: What is `start_of ` used for?

Comment (Ed-XCF) [reply bali/utils/timezone.py:84]: `start_of` is clearer and has more functions  then `get_beginning_datetime`

Comment (Ed-XCF) [reply bali/utils/timezone.py:83]: and we can define `end_of` base on `start_of` in future

**Pull request merged**